pkg/cli/mapping_server.go

```diff
@@ -117,15 +129,31 @@ func (c *MappingServerCommand) RunUnstarted(ctx context.Context, args []string)
 		return nil, nil, closer, fmt.Errorf("server.NewHandler: %w", err)
 	}
 
+	srv, err := serving.New(c.cfg.Port)
+	if err != nil {
+		return nil, nil, closer, fmt.Errorf("failed to create serving infrastructure: %w", err)
+	}
+
 	closer = func() {
-		if err := handler.Cleanup(); err != nil {
-			logger.Errorw("failed to close clean up handler", "error", err)
+		successTopic.Stop()
+		failureTopic.Stop()
+
+		if err := pubsubClient.Close(); err != nil {
+			logger.Errorw("failed to close pubsub client", "error", err)
+		}
+
+		if err := assetClient.Close(); err != nil {
+			logger.Errorw("failed to stop asset client", "error", err)
 		}
 	}
```

Comment:
Move to be before `srv, err := serving.New(c.cfg.Port)`

@sqin2019 - I thought about what we discussed offline. In case, we return err, `Run()` will not call close func. This is ok because we haven't started serving traffic. 

I agree it would be better if we can close things properly when error happens. But let's fix that separately.

Reply:
sg